File: 1500-count-largest-group/count-largest-group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        counter = Counter()

        for i in range(1, n + 1):
            key = sum([int(x) for x in str(i)])
            counter[key] += 1

        mxVal = max(counter.values())
        count = 0

        for val in counter.values():
            if val == mxVal:
                count += 1

        return count

File: 1500-count-largest-group/count-largest-group.js
/**
 * @param {number} n
 * @return {number}
 */
var countLargestGroup = function (n) {
    const counter = new Map();

    for (let i = 1; i <= n; i++) {
        const key = i.toString().split('').reduce((sum, digit) => sum + Number(digit), 0);
        counter.set(key, (counter.get(key) || 0) + 1);
    }

    let maxCount = Math.max(...counter.values());
    let count = 0;

    for (let val of counter.values()) {
        if (val === maxCount) {
            count++;
        }
    }

    return count;
};